src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -336,6 +343,15 @@ private ClusterDescription ProcessReplicaSetChange(ClusterDescription clusterDes
                             {
                                 var server = _servers.SingleOrDefault(x => EndPointHelper.Equals(args.NewServerDescription.EndPoint, x.EndPoint));
                                 server.Invalidate();
+                                
+                                _sdamInformationEventHandler?.Invoke(new SdamInformationEvent(() =>
```

Comment:
Equivalent lines in Java: https://github.com/mongodb/mongo-java-driver/blob/3e777058b556d67c68b084372f940f80eb0e9398/driver-core/src/main/com/mongodb/internal/connection/MultiServerCluster.java#L240-L244